lab3/main.cpp

```diff
@@ -0,0 +1,153 @@
+#include <iostream>
+#include <stdlib.h>
+#include <time.h>
+
+enum Command { RandomlyFillMatrix = 0, ManuallyFillMatrix = 1 };
+
+// Using **& in the functions will make it more memory efficient
+
+void FreeMemory(int **&matrix, int N) {
+    for (int i = 0; i < N; ++i) {
+        delete[] matrix[i];
+    }
+    delete[] matrix;
+}
+
+int GetMatrixElement(int **&matrix, int i, int j, int N) {
+    // The sum is bigger than N for elements in the second triangle
+    if (i + j >= N) {
+        return matrix[N - 1 - j][N - 1 - i];
+    } else
+        return matrix[i][j];
+}
+
+void AllocateMemory(int **&target, int N) {
+    target = new int *[N];
+    for (int i = 0; i < N; i++) {
+        target[i] = new int[N - i];
```

Comment:
Why N-1 if you need to store only triangle of data

Reply:
> Why N-1 if you need to store only triangle of data
`target[i] = new int[N - i];`
It's more like `N-i`, not `N-1`, I guess.